### ISSUE TITLE
Telemetry distancecompleted can be -2147483648

### ISSUE THREAD

**Issue by kmadsen** [mapbox/mapbox-navigation-android#2748]
**Mapbox Navigation SDK version:** 1.0.0

### Steps to trigger behavior

https://app.mode.com/editor/mapbox/reports/36579db91d6d

distancecompleted is incorrect in telemetry. 

![Screen Shot 2020-04-07 at 6 07 26 PM](https://user-images.githubusercontent.com/3021882/78733580-af835b80-78fa-11ea-958a-27e1a1fe2f1d.png)

### Expected behavior
distancecompleted = 0 to estimateddistance

### Actual behavior
distancecompleted can be -2147483648 (aka MIN_INTEGER)

**Comment by kmadsen** [comment]
Have verified that this is not 100% reproducible. Through testing have seen that `distancecompleted` is calculated accurately

**Comment by kmadsen** [comment]
Added a query to see how often this happens per session. Technically, it can only be reproduced 20% of the time 😢 😆 

| error sessions | total sessions | success_rate |
| -- | -- | -- |
| 34 | 177 | 80.8%

**Comment by kmadsen** [comment]
Found a likely cause for this issue. It's when you arrive at your destination and continue moving

```
D/MAPBOX_TELEMETRY: ARRIVAL event sent true
arrival_debug distanceRemaining 5.92887
arrival_debug distanceRemaining -9.536743E-7
arrival_debug distanceRemaining -9.536743E-7
arrival_debug distanceRemaining -9.536743E-7
arrival_debug distanceRemaining -9.536743E-7
arrival_debug distanceRemaining -9.536743E-7
arrival_debug distanceRemaining -9.536743E-7
```

**Comment by Guardiola31337** [comment]
Commenting here so this ticket is not automatically closed by `stale bot` as it's still relevant.

**Comment by Guardiola31337** [comment]
Where are we here @kmadsen? Is this still an issue?

Refs. https://github.com/mapbox/mapbox-navigation-android/issues/3164

**Comment by kmadsen** [comment]
re-ran the query and the issue is still happening on 1tap. fair to say it's still an issue
https://app.mode.com/editor/mapbox/reports/36579db91d6d

useragent | created | event | session | distanceremaining | originalestimateddistance | estimateddistance | distancecompleted
-- | -- | -- | -- | -- | -- | -- | --
com.mapbox.onetap.app/v0.17.0/998351 mapbox-navigation-android/1.0.0-rc.7 | 2020-09-09 7:53:52 | navigation.cancel | aace88fe-47fb-4aa2-8d6c-f883b746addf | 56 | 6152 | 484 | -2147483648
com.mapbox.onetap.app/v0.17.0/998351 mapbox-navigation-android/1.0.0-rc.7 | 2020-09-09 7:53:51 | navigation.reroute | aace88fe-47fb-4aa2-8d6c-f883b746addf | 56 | 6152 | 484 | -2147483648
com.mapbox.onetap.app/v0.17.0/998351 mapbox-navigation-android/1.0.0-rc.7 | 2020-09-09 7:53:49 | navigation.reroute | aace88fe-47fb-4aa2-8d6c-f883b746addf | 56 | 6152 | 484 | -2147483648
com.mapbox.onetap.app/v0.17.0/998351 mapbox-navigation-android/1.0.0-rc.7 | 2020-09-09 7:53:38 | navigation.feedback | aace88fe-47fb-4aa2-8d6c-f883b746addf | 484 | 6152 | 484 | -2147483648

**Comment by korshaknn** [comment]
@kmadsen @Guardiola31337 
hey! going to fix it during telemetry refactoring. how do we calculate `distanceCompleted`?
I just use
`distanceCompleted = routeProgress.distanceTraveled.toInt()`
is it valid ^^ ?
should we have any additional calculations for offRoute event?

**Comment by Guardiola31337** [comment]
@korshaknn 
> how do we calculate `distanceCompleted`?
I just use
`distanceCompleted = routeProgress.distanceTraveled.toInt()`
is it valid ^^ ?
should we have any additional calculations for offRoute event?

That should be handled by NN. We recently started relying on whatever is sent from NN https://github.com/mapbox/mapbox-navigation-android/blob/c2e0546074d844db6aa585b6906242e1aae355a8/libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/NavigatorMapper.kt#L122-L123 https://github.com/mapbox/mapbox-navigation-android/pull/3604 (previously we some of the computations were done inside the SDK code). Can we retest and see if this still an issue? This landed in [`v1.2.0-alpha.1`](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v1.2.0-alpha.1).

cc @kmadsen

**Comment by kmadsen** [comment]
> how do we calculate distanceCompleted?
https://github.com/mapbox/mapbox-navigation-android/blob/c5b8f6185ac1f22262664f14167da2d7ef2522e2/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt#L632-L635

Adding to pablo's comment cause here is where distanceCompleted is literally set

Rerunning the query that's in the description: https://app.mode.com/editor/mapbox/reports/36579db91d6d
``` sql
SELECT 
  event,
  count(distinct CASE WHEN (distancecompleted = -2147483648) THEN sessionidentifier ELSE NULL END) as error_sessions,
  count(distinct sessionidentifier) as total_sessions
FROM sdk_events.combined_events 
WHERE  cast(dt AS timestamp) > NOW() - INTERVAL '5' DAY
  and sdkversion like '1.%'
  and useragent like 'com.mapbox.onetap.app%'
GROUP BY event
```

![Screen Shot 2020-10-23 at 11 29 03 AM](https://user-images.githubusercontent.com/3021882/97040600-04975f80-1523-11eb-8480-6eb6f7b1cb7c.png)

This appears fixed! 🥳 . We also would like to verify with thousands of sessions, where we are still in the hundreds. We can re-open this when/if we see it again.
